src/Livewire/Accounting/PaymentReminder.php

```diff
@@ -53,30 +60,114 @@ protected function getSelectedActions(): array
+    }
+
+    protected function getCommunicatableId(OffersPrinting $item): int
+    {
+        return $item->order_id;
+    }
+
+    protected function getTo(OffersPrinting $item, array $documents): array
+    {
+        return Arr::wrap($item->getPaymentReminderText()?->mail_to
+            ?: $item->order->contact->invoiceAddress->email_primary
+        );
+    }
+
+    protected function getCc(OffersPrinting $item): array
+    {
+        return Arr::wrap($item->getPaymentReminderText()?->mail_cc);
+    }
+
+    protected function getCreateAttachmentArray(OffersPrinting $item, string $view): array
+    {
+        return [
+            'model_type' => $item->getMorphClass(),
+            'model_id' => $item->getKey(),
+            'view' => $view,
+            'name' => __($view),
+            'attach_relation' => 'order',
         ];
     }
 
-    public function createNextPaymentReminder(): void
+    protected function getSubject(OffersPrinting $item): string
+    {
+        return html_entity_decode($item->getPaymentReminderText()?->mail_subject ?? '') ?:
+            __(
+                'Payment Reminder :level for invoice :invoice-number',
+                [
+                    'level' => $item->reminder_level,
+                    'invoice-number' => $item->order->invoice_number,
+                ]
+            );
+    }
+
+    protected function getHtmlBody(OffersPrinting $item): string
+    {
+        return html_entity_decode(
+            $item->getPaymentReminderText()?->mail_body
+                    ?? $item->getPaymentReminderText()?->reminder_body
+                    ?? ''
+        );
+    }
+
+    protected function getBladeParameters(OffersPrinting $item): array|SerializableClosure|null
+    {
+        return new SerializableClosure(
+            fn () => ['paymentReminder' => app(PaymentReminderModel::class)->whereKey($item->getKey())->first()]
+        );
+    }
+
+    protected function getPrintLayouts(): array
+    {
+        return [
+            'payment-reminder',
+        ];
+    }
+
+    public function createDocuments(): null|MediaStream|Media
     {
         $orders = $this->getSelectedModels();
 
-        $mailMessages = [];
+        try {
+            resolve_static(CreatePaymentReminder::class, 'canPerformAction', [true]);
+        } catch (UnauthorizedException $e) {
+            exception_to_notifications($e, $this);
+
+            return null;
+        }
+
+        $documents = collect();
+        $reminderTextExists = [];
         foreach ($orders as $order) {
             try {
                 $paymentReminder = CreatePaymentReminder::make([
                     'order_id' => $order->id,
                 ])
-                    ->checkPermission()
                     ->validate()
                     ->execute();
 
-                $paymentReminderText = app(PaymentReminderText::class)
+                $reminderTextExists[$paymentReminder->reminder_level] ??= app(PaymentReminderText::class)
```

Comment:
resolve_static()

Reply:
had to add a line break, looks ugly but no better idea here